### PR TITLE
Fidelianawar/msal test with ciam

### PR DESCRIPTION
**Pull request by fidelianawar**
## Proposed changes

Perform requested engineering updates needed to support CIAM scenarios (acquireTokenInteractive and acquireTokenSilent) for MSAL. ([Parent PBI](https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2069351))

## Type of change

- [ ] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [X] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [X] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: MSAL/test/automation/tests/interactive/MSALCIAMTest.m [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    [self loadTestApp:appConfigurationRequest];
+}
+
+#pragma mark - Tests
+
+-
+    (void)testInteractiveAndSilentCIAMLogin_withPromptAlways_noLoginHint_andSystemWebView
+{
+    MSIDAutomationTestRequest *request = [self.class.confProvider defaultAppRequest:self.testEnvironment targetTenantId:self.primaryAccount.targetTenantId];
+    request.configurationAuthority = [self.class.confProvider defaultAuthorityForIdentifier:self.testEnvironment tenantId:@"organizations"];
```
Comment (oldalton) [inline MSAL/test/automation/tests/interactive/MSALCIAMTest.m:65]: Why are we going to organizations endpoint here?

File: MSAL/test/automation/tests/interactive/MSALCIAMTest.m [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+-
+    (void)testInteractiveAndSilentCIAMLogin_withPromptAlways_noLoginHint_andSystemWebView
+{
+    MSIDAutomationTestRequest *request = [self.class.confProvider defaultAppRequest:self.testEnvironment targetTenantId:self.primaryAccount.targetTenantId];
+    request.requestScopes = @"https://substrate.office.com/profile//User.Read";
+    request.promptBehavior = @"force";
+    request.redirectUri = @"msauth.com.microsoft.msalautomationapp://auth";
+    
+    // 1. Do interactive login
+    NSString *homeAccountId = [self runSharedADFSInteractiveLoginWithRequest:request];
```
Comment (oldalton) [inline MSAL/test/automation/tests/interactive/MSALCIAMTest.m:70]: Why does this test subclass ADFS tests? Should it subclass AAD tests instead?

File: MSAL/test/automation/tests/interactive/MSALCIAMTest.m [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+- (void)setUp
+{
+    [super setUp];
+    
+    self.testEnvironment = self.class.confProvider.wwEnvironment;
+    
+    MSIDTestAutomationAccountConfigurationRequest *accountConfigurationRequest = [MSIDTestAutomationAccountConfigurationRequest new];
+    accountConfigurationRequest.environmentType = self.testEnvironment;
+    accountConfigurationRequest.federationProviderType = @"ciam";
```
Comment (Veena11) [inline MSAL/test/automation/tests/interactive/MSALCIAMTest.m:46]: could you define this in the MSIDTestAutomationAccountConfigurationRequest along with other FederationProviderTypes

File: MSAL/test/automation/tests/interactive/MSALCIAMTest.m [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    [self loadTestApp:appConfigurationRequest];
+}
+
+#pragma mark - Tests
+
+-
+    (void)testInteractiveAndSilentCIAMLogin_withPromptAlways_noLoginHint_andSystemWebView
+{
+    MSIDAutomationTestRequest *request = [self.class.confProvider defaultAppRequest:self.testEnvironment targetTenantId:self.primaryAccount.targetTenantId];
+    request.requestScopes = @"https://substrate.office.com/profile//User.Read";
```
Comment (rayluo) [inline MSAL/test/automation/tests/interactive/MSALCIAMTest.m:66]: Can we not hard code this scope? It is available from one of the lab apis (the second one) [mentioned here](https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/520#issue-1487352636). In the future, we may set up a different scope in the lab. At that time, this test case would likely break.

**Review by oldalton — APPROVED** [review]

**Pull request merged**